docs/guide/upgrading.md

```diff
@@ -8,29 +8,30 @@ Apostrophe 3 introduces a number of new features for developers and content-edit
 
 ### 100% RESTful API coverage
 
-In the previous version, developers who wanted REST API access to their data needed to install the `apostrophe-headless` module. [These are now built into A3](/reference/api/) and are the foundation of the user interface.
+In the previous version, developers who wanted REST API access to their data needed to install the `apostrophe-headless` module. [This is now built into A3](/reference/api/) and are the foundation of the user interface.
 
 ### Totally rebuilt content editing interface
 
 The UI has been completely rebuilt with [Vue.js](https://vuejs.org/). It is much faster, more accessible, and easier to configure. It will also stay out of your way, so you can build with React, vanilla JavaScript, a different Vue.js version, or any other framework without any issue.
 
 ### Clearer and easier module building
 
-There were several things about building advanced modules in A2 that often tripped up developers. A3 helps avoid the confusion around where to put things by doing away with `beforeConstruct`, `construct`, and `afterConstruct`. [They are replaced by a single `init` function and special sections](/reference/module-api/module-overview.md#initialization-function) to do things like extend API routes, add template helper methods, and add columns to the piece manager UI.
+There were several things about building advanced modules in A2 that often tripped up developers. A3 helps avoid the confusion around where to put things by doing away with `beforeConstruct`, `construct`, and `afterConstruct`. [They are replaced by a single `init` function and special sections](/reference/module-api/module-overview.md#initialization-function) to do things like extend API routes, add template helper methods, and add columns to the piece manager UI. The `addFields` array also became a `fields` object with an `add` subproperty to prevent problems with common inheritance patterns.
```

Comment:
`fields` can optionally be a customization function too, but OK, don't want to blow their minds too hard at this stage.